### PR TITLE
Remove dag.run() method

### PR DESCRIPTION
**Pull request by dstandish**
This method uses Backfill internally.  Before we can remove BackfillJobRunner, we need to remove DAG.run.  But before we can remove DAG.run, we need to update some old tests that use it.  So this is the first step towards removing BackflilJobRunner.

There were some very old tests that came from airflow github issue 1225.  These appeared to test the scheduler but really they tested the backfill job runner.  Just to be cautious, I kept most of them rather than remove (which probably would have been fine since they essentially tested code that we'll be removing).  As appropriate I either changed them to run on dag.test or scheduler.  The ones dealing with ignore first depends on past will have to be added back when that functionality is implemented in new backfill.

### CODE REVIEW

**Comment by potiuk** [comment]
🧛

File: airflow/example_dags/example_bash_operator.py [hunk 1]
```diff
@@ -73,5 +73,10 @@
 # [END howto_operator_bash_skip]
 this_will_skip >> run_this_last
 
+from tests.system.utils import get_test_run  # noqa: E402
```
Comment (potiuk) [inline airflow/example_dags/example_bash_operator.py]: Actually those lines are still needed I think. They are also using "dag.test" under the hood.

Comment (potiuk) [reply airflow/example_dags/example_bash_operator.py]: (maybe we can rename them, but they are still neded to use pytest).

Comment (dstandish) [reply airflow/example_dags/example_bash_operator.py]: i'm confused -- you say the lines are still needed but i am not removing anything here 😕 ?

Comment (potiuk) [reply airflow/example_dags/example_bash_operator.py]: OH GOD .... 

Yeah.. Somehow in my head those additions were .... removals :)

Friday night after returning back from holidays and having cold whole week 🤯

Comment (dstandish) [reply airflow/example_dags/example_bash_operator.py]: lol know the feeling

File: providers/src/airflow/providers/google/marketing_platform/example_dags/example_display_video.py [hunk 2]
```diff
@@ -125,7 +125,7 @@
     "example_display_video_sdf",
     start_date=START_DATE,
     catchup=False,
-) as dag3:
+) as dag_example_display_video_sdf:
```
Comment (ashb) [inline providers/src/airflow/providers/google/marketing_platform/example_dags/example_display_video.py:128]: Nit: unrelated changes in this file

File: providers/tests/google/cloud/operators/test_datastore_system.py [hunk 3]
```diff
@@ -44,8 +44,8 @@ def teardown_method(self):
 
     @provide_gcp_context(GCP_DATASTORE_KEY)
     def test_run_example_dag(self):
-        self.run_dag("example_gcp_datastore", CLOUD_DAG_FOLDER)
+        self.run_dag("example_gcp_datastore", CLOUD_DAG_FOLDER)  # this dag does not exist?
 
     @provide_gcp_context(GCP_DATASTORE_KEY)
     def test_run_example_dag_operations(self):
-        self.run_dag("example_gcp_datastore_operations", CLOUD_DAG_FOLDER)
+        self.run_dag("example_gcp_datastore_operations", CLOUD_DAG_FOLDER)  # this dag does not exist?
```
Comment (ashb) [inline providers/tests/google/cloud/operators/test_datastore_system.py:51]: How on earth have these tests been passing? I guess they haven't as we don't regularly run these system tests? 

These dags were removed/moved/renamed in https://github.com/apache/airflow/pull/23108 by @bhirsz.

I guess the google system tests can't have been fully run since 2022 then. Yay

File: providers/tests/google/cloud/transfers/test_facebook_ads_to_gcs_system.py [hunk 4]
```diff
@@ -26,12 +26,13 @@
 from airflow.exceptions import AirflowException
 from airflow.models import Connection
 from airflow.utils.process_utils import patch_environ
+from providers.src.airflow.providers.google.cloud.example_dags import example_facebook_ads_to_gcs
```
Comment (ashb) [inline providers/tests/google/cloud/transfers/test_facebook_ads_to_gcs_system.py]:
```suggestion
from airflow.providers.google.cloud.example_dags import example_facebook_ads_to_gcs
```

(And then imports'll need re-ordering)

File: providers/tests/system/google/cloud/dataprep/example_dataprep.py [hunk 5]
```diff
@@ -313,6 +313,13 @@ def delete_connection(connection_id: str) -> None:
     # when "tearDown" task with trigger rule is part of the DAG
     list(dag.tasks) >> watcher()
 
+
+from dev.tests_common.test_utils.system_tests import get_test_run  # noqa: E402
+
+# Needed to run the example DAG with pytest (see: tests/system/README.md#run_via_pytest)
+test_run = get_test_run(dag)
```
Comment (ashb) [inline providers/tests/system/google/cloud/dataprep/example_dataprep.py]: Something in the pre-commit made this a duplicated block -- it already exists a few lines below.

**Review by ashb — COMMENTED** [review]
Few nits/small changes.

And we should fix those broken google system test dags, or perhaps since they clearly couldn't have been being run remove them entirely?

File: providers/tests/google/cloud/transfers/test_facebook_ads_to_gcs_system.py (hunk 4 above)
Comment (ashb) [reply providers/tests/google/cloud/transfers/test_facebook_ads_to_gcs_system.py]: It might be worth adding `providers.src.` to the forbidden import list for ruff inside `pyproject.toml`

File: providers/src/airflow/providers/google/marketing_platform/example_dags/example_display_video.py (hunk 2 above)
Comment (dstandish) [reply providers/src/airflow/providers/google/marketing_platform/example_dags/example_display_video.py:128]: it's so that they can be imported in a test 

see providers/tests/google/marketing_platform/operators/test_display_video_system.py

File: providers/tests/google/cloud/operators/test_datastore_system.py (hunk 3 above)
Comment (dstandish) [reply providers/tests/google/cloud/operators/test_datastore_system.py:51]: i don't think we run them. we just spend a lot of time updating them in this PR :)

File: providers/tests/system/google/cloud/dataprep/example_dataprep.py (hunk 5 above)
Comment (dstandish) [reply providers/tests/system/google/cloud/dataprep/example_dataprep.py]: resolved

File: providers/tests/google/cloud/transfers/test_facebook_ads_to_gcs_system.py (hunk 4 above)
Comment (dstandish) [reply providers/tests/google/cloud/transfers/test_facebook_ads_to_gcs_system.py]: leaving for another day, just need to this PR and get onto other ones

File: airflow/models/dag.py [hunk 6]
```diff
@@ -2296,84 +2293,8 @@ def _remove_task(self, task_id: str) -> None:
...
-            verbose=verbose,
-            conf=conf,
-            rerun_failed_tasks=rerun_failed_tasks,
-            run_backwards=run_backwards,
-            run_at_least_once=run_at_least_once,
-            continue_on_failures=continue_on_failures,
-            disable_retry=disable_retry,
-        )
-        run_job(job=job, execute_callable=job_runner._execute)
+    def run(self, *args, **kwargs):
```
Comment (kaxil) [inline airflow/models/dag.py:2296]: Isn't it easier to just remove this too?

**Review by kaxil — APPROVED** [review]

File: airflow/models/dag.py (hunk 6 above)
Comment (dstandish) [reply airflow/models/dag.py:2296]: there are things that mock it as part of testing of backfill. but that is easier to remove when we will remove / replace legacy backfill.  but doing that in this PR together is just too much IMO
